proofgeneral/narya.el

```diff
@@ -197,10 +230,28 @@ handling in Proof General."
         (parsed-hole-data nil)
         (error-found nil)
         (reformatted nil))
+    (when (and span (overlay-buffer span))
+    ;; Store the span
+    (setq narya-last-successful-span span))
     ;; Check for errors in the output first.
-    (when (string-match proof-shell-error-regexp string)
-      (setq error-found t
-            proof-shell-last-output-kind 'error))
+    (when (and (string-match proof-shell-error-regexp string))
+        (setq error-found t
+              proof-shell-last-output-kind 'error)
+        (setq span narya-last-successful-span)
+        (proof-with-script-buffer
+   	;; Start parsing the error numbers after the initial "^L[errors]^L" part
+        (let ((error-start (string-match "\f\\[errors\\]\f\n" string)))
+           ;; Parse all the error pairs (start-byte, end-byte)
+           (while (string-match "\\([0-9]+\\) \\([0-9]+\\)" string)
+              (let ((start-byte (string-to-number (match-string 1 string)))
+                    (end-byte (string-to-number (match-string 2 string))))
+              ;; Highlight the positions on the current line
+              (if (and span (overlay-buffer span) (overlay-end span)) 
+                  (narya-highlight-error-range (+ (overlay-end span) 1 start-byte) 
+                                               (+ (overlay-end span) 1 end-byte))
```

Comment:
```suggestion
                  (narya-highlight-error-range (+ (position-bytes (overlay-end span)) 1 start-byte) 
                                               (+ (position-bytes (overlay-end span)) 1 end-byte))
```
The overlay-end is a character position; you have to convert it to a byte position before adding it to the Narya-supplied byte offsets.